### PR TITLE
Password show toggle cursor state preserved on Chrome #12018

### PR DESCRIPTION
**Pull request by esh-g**
### Details

This change fixes the cursor resetting behavior on Chrome while toggling the password show/hide button. Chrome has a weird bug that requires code to be executed on the `onMouseDown` instead of `onPress`

### Fixed Issues

$ https://github.com/Expensify/App/issues/12018
$ https://github.com/Expensify/App/issues/12018#issuecomment-1314225656

### Tests

1. Focus the password input on login screen
2. Add some text to the input
3. Click on the eye icon
4. Verify that the cursor preserves it position after toggle
- [x] Verify that no errors appear in the JS console

### Offline tests

When offline, the same steps can be followed and the result is also same.

1. Focus the password input on login screen
2. Add some text to the input
3. Click on the eye icon
4. Verify that the cursor preserves it position after toggle

This issue was specific to chrome only as discussed here:  https://github.com/Expensify/App/issues/12018#issuecomment-1310165970

### QA Steps

1. Focus the password input on login screen
2. Add some text to the input
3. Click on the eye icon
4. Verify that the cursor preserves it position after toggle

This issue was specific to chrome only as discussed here:  https://github.com/Expensify/App/issues/12018#issuecomment-1310165970

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ x I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
 Note: For chrome specific. Here is the video: 
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
    ### Note: This issue was only visible on chrome as mentioned here: https://github.com/Expensify/App/issues/12018#issuecomment-1310165970
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for expected offline behavior are in the `Offline steps` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] I verified that if a function's arguments changed that all usages have also been updated correctly
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/77237602/203388521-206dfb79-bb89-4e96-a55c-cdec446ab940.mp4

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/77237602/203406492-b245f46e-4ed6-49f0-96e7-a22482f91a8c.mp4

### CODE REVIEW

**Comment by Someone-here** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Julesssss** [comment]
@Someone-here please review and add screenshots for all platforms. This is to help prevent regressions that can occur.

**Comment by eVoloshchak** [comment]
@Someone-here 
Bug:
If you press and hold on the right part of the input (not the icon itself), it still resets cursor to start

https://user-images.githubusercontent.com/9059945/203843596-07ebc9cc-d1a0-4b51-99e8-81ffcefea73f.mov

**Comment by Someone-here** [comment]
I think this is due to having both `onPress` and `onMouseDown` handlers, so I think for web platforms, I need to pass only the `onMouseDown` but `onPress` for native apps. That will be platform specific code.

**Comment by Someone-here** [comment]
What do you think? Should I create a wrapper component that checks the platform and passes the appropriate prop according to platform?

**Comment by eVoloshchak** [comment]
> What do you think? Should I create a wrapper component that checks the platform and passes the appropriate prop according to platform?

No need, there is already index.js and index.native.js files for `BaseTextInput`
You can pass the prop there

**Comment by eVoloshchak** [comment]
Also please make sure to

1. Fill out the PR Author checklist so it doesn't fail
2. Test on **all platforms** and upload the screen recordings for them. Even if this bug is specific to Chrome, we want to make sure there are no regressions on other platforms
3. If you can't test iOS and MacOS - i can help with that. But next time please read  [TESTING_MACOS_AND_IOS.md](https://github.com/Expensify/App/blob/main/contributingGuides/TESTING_MACOS_AND_IOS.md) to use cloud services that offer MacOS instances.

File: src/components/TextInput/BaseTextInput.js [hunk 1]
```diff
@@ -292,7 +292,10 @@ class BaseTextInput extends Component {
                                         <Checkbox
                                             style={styles.secureInputShowPasswordButton}
                                             onPress={this.togglePasswordVisibility}
-                                            onMouseDown={e => e.preventDefault()}
+                                            onMouseDown={e => {
```
Comment (eVoloshchak) [inline src/components/TextInput/BaseTextInput.js:295]: Please fix linter errors

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by Someone-here** [comment]
Okay, I understand. Will do as you say. Sorry for the inconvenience, this is my first time contributing to expensify therefore I hope you understand this lacking.

**Comment by eVoloshchak** [comment]
> Okay, I understand. Will do as you say. Sorry for the inconvenience, this is my first time contributing to expensify therefore I hope you understand this lacking.

This is absolutely normal, no worries :)

**Comment by Someone-here** [comment]
I have fixed the issue that you mentioned that it resets when clicking outside the eye as well. And this solution doesn't even require native file for checkbox. I borrowed a Mac from a friend and ran the tests on iOS and Safari as well. It runs just as expected.

Will share the videos for that shortly.

**Comment by Someone-here** [comment]
Also added space key support for toggling the password

**Comment by Someone-here** [comment]
Here are the testing videos:

### ANDROID

https://user-images.githubusercontent.com/77237602/204093209-2e1ae079-ac3b-4943-b513-64c4ddb0350b.mp4

### iOS

https://user-images.githubusercontent.com/77237602/204093257-e964fcbf-6519-487a-9b1a-7f7247ef12d8.mp4

### Safari

https://user-images.githubusercontent.com/77237602/204093285-0e168bd4-ac44-48fc-ba7a-fd60ffbf31b4.mp4

### Chrome (mobile)

https://user-images.githubusercontent.com/77237602/204093531-b041af1f-246e-46dc-ac8e-5c30e8a4248d.mp4


### Chrome (desktop)

https://user-images.githubusercontent.com/77237602/204093312-31072d45-b573-45ae-8021-7ce44098560b.mp4

File: src/components/TextInput/BaseTextInput.js [hunk 2]
```diff
@@ -186,7 +186,9 @@ class BaseTextInput extends Component {
         ]).start();
     }
 
-    togglePasswordVisibility() {
+    togglePasswordVisibility(e) {
+        if (!e) return;
```
Comment (eVoloshchak) [inline src/components/TextInput/BaseTextInput.js:190]:
```suggestion
        if (!event) {
           return;
        }
        event.preventDefault();
```

Comment (eVoloshchak) [reply src/components/TextInput/BaseTextInput.js:190]: Lets change this so it matches the styling with the rest of the app

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
> Fixed the styling again. Does it work as expected?

nice, testing!

**Comment by eVoloshchak** [comment]
@Someone-here 
I can still reproduce the bug in Chrome


https://user-images.githubusercontent.com/9059945/204351222-85982ec2-3e4e-43be-890b-aced613765a5.mov

**Comment by Someone-here** [comment]
Could you please share the chrome version that you are using?

**Comment by eVoloshchak** [comment]
> Could you please share the chrome version that you are using?

Version 107.0.5304.110 (Official Build) (arm64)

**Comment by Someone-here** [comment]
I was testing on chrome 107 (64-bit) and it seems to work there flawlessly. Let me investigate this on mac

**Comment by eVoloshchak** [comment]
> I was testing on chrome 107 (64-bit) and it seems to work there flawlessly. Let me investigate this on mac

I'd also recommend you to pull the latest changes from main, that's how i tested

**Comment by Someone-here** [comment]
Okay, I have a solution.
The issue is that the two events `onPress` and `onMouseDown` sometimes get called together and this is what causes the issue. 

I thought I could fix it without using native file but I'm afraid I would have to create a native file for `checkBox.js` as `checkBox.native.js`. And make sure to pass the right props according to the platform. 
Does that solution sound good?

Are there any additional steps to register the native file or are they automatically recognised by the filename?

**Comment by eVoloshchak** [comment]
Yup, that sounds good!
There are no additional steps, all that's needed is proper naming

**Pull request closed without merging**